BEIMA.Backend.Test/MongoService/MongoConnectorTest.cs

```diff
@@ -160,5 +160,121 @@ public void ConnectorCreated_CheckIsConnected_TrueReturned()
             var result = mongo.IsConnected();
             Assert.IsTrue(result);
         }
+
+        //DeviceType tests
+
+        [Test]
+        public void ConnectorCreated_GetDeviceTypeGivenValidDeviceTypeId_DeviceDocumentReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //Setup (inserting a document)
+            var doc = new BsonDocument {
+                { "name", "TestDeviceType"},
+                { "description", "This is a test" }
+            };
+            var insertResult = mongo.InsertDeviceType(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Test (retrieve the document)
+            var retrievedDoc = mongo.GetDeviceType((ObjectId)doc["_id"]);
+            Assert.IsNotNull(retrievedDoc);
+            Assert.That(retrievedDoc, Is.TypeOf(typeof(BsonDocument)));
+        }
+
+        [Test]
+        public void ConnectorCreated_GetAllDeviceTypes_DeviceTypeDocumentListReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //Setup (inserting a document)
+            var doc = new BsonDocument {
+                { "name", "TestDeviceType"},
+                { "description", "This is a test" }
+            };
+            var insertResult = mongo.InsertDeviceType(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Test (retrieve all documents)
+            var retrievedDocs = mongo.GetAllDeviceTypes();
+            Assert.IsNotNull(retrievedDocs);
+            Assert.That(retrievedDocs, Is.TypeOf(typeof(List<BsonDocument>)));
+        }
+
+        [Test]
+        public void DeviceTypeNotInserted_InsertDeviceTypeAndDeleteDeviceType_DeviceTypeHasBeenDeleted()
+        {
+            var mongo = MongoConnector.Instance;
+            var doc = new BsonDocument {
+                { "name", "TestDeviceType"},
+                { "description", "This is a test" }
+            };
+            //Insert device type
+            var insertResult = mongo.InsertDeviceType(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
```

Comment:
Since this test is testing the delete functionality, we don't need asserts on if the inserting the device type was successful

Reply:
Same here, you can use Assume instead of Assert for semantic purposes.

Reply:
Fixed [bb6c78f](https://github.com/cs481-ekh/s22-beima/pull/134/commits/bb6c78f547b5e677d7518b3bf4db83c584eb42d7)